src/scripts/components/widgets/VolumeBar.js

```diff
@@ -0,0 +1,133 @@
+/* @flow */
+
+import React, { Component } from 'react'
+import styled from 'styled-components'
+
+type Props = {
+  onVolumeChange: (percentage: number) => void,
+  currentVolume: number
+}
+
+type State = {
+  userIsDragging: boolean,
+  draggingVolumePercentage: number
+}
+
+const VOLUME_INDICATOR_DIMENSION: number = 20
+
+const VolumeIndicator = styled.div`
+  position: absolute;
+  width: ${VOLUME_INDICATOR_DIMENSION}px;
+  height: ${VOLUME_INDICATOR_DIMENSION}px;
+  border-radius: 50%;
+  background-color: ${({ theme }) =>
+    theme.colors.VideoPlayer.progress.scrubber};
+`
+
+/* prettier-ignore */
+const VolumeBar = styled.div`
+  width: 100;
+  height: 5px;
+  position: relative;
+  display: flex;
+  justify-content: flex-end;  
+  align-items: center;
+  background: linear-gradient(to right, ${({ theme }) => `${theme.colors.VideoPlayer.progress.barFrom}, ${theme.colors.VideoPlayer.progress.barTo}`});
+  ${/* sc-selector */VolumeIndicator} {
+    left: ${({ currentVolume, draggingVolumePercentage }) => {
+    if (draggingVolumePercentage) {
+      return `calc(${Math.max(0, Math.min(100, draggingVolumePercentage))}% - ${VOLUME_INDICATOR_DIMENSION / 2}px)`
+    }
+    return `calc(${Math.max(0, Math.min(100, currentVolume))}% - ${VOLUME_INDICATOR_DIMENSION / 2}px)`
+  }};
+  }
+  `
+
+class PlayerControls extends Component<Props, State> {
+  volumeBarRef: ?HTMLElement
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {
+      userIsDragging: false,
+      draggingVolumePercentage: 0
+    }
+
+    this.addMouseEventListeners()
+  }
+
+  componentWillUnmount (): void {
+    this.removeMouseEventListeners()
+  }
+
+  onMouseUp = (e: Object): void => {
+    this.setState((prevState: State) => {
+      if (prevState.userIsDragging) {
+        e.stopPropagation()
+        return {
+          userIsDragging: false,
+          draggingVolumePercentage: 0
+        }
+      }
+    })
+  }
+
+  onMouseMove = (e: Object): void => {
```

Comment:
There is perhaps some "dragging/scrubbing" code that is in both this component and the player time scrubber that could be DRY'd up and abstracted. Something to consider for the future, especially if we need a third scrubbing bar somewhere in the UI